src/main/java/org/gridsuite/modification/server/modifications/GeneratorModification.java

```diff
@@ -170,9 +170,17 @@ private void modifyGeneratorReactiveCapabilityCurvePoints(GeneratorModificationI
     private Reporter modifyGeneratorActiveLimitsAttributes(GeneratorModificationInfos modificationInfos,
                                                            Generator generator, Reporter subReporter) {
         Reporter subReporterLimits = null;
+        Report reportMaxActivePower;
+        Report reportMinActivePower;
 
-        Report reportMaxActivePower = ModificationUtils.getInstance().applyElementaryModificationsAndReturnReport(generator::setMaxP, generator::getMaxP, modificationInfos.getMaxActivePower(), "Max active power");
-        Report reportMinActivePower = ModificationUtils.getInstance().applyElementaryModificationsAndReturnReport(generator::setMinP, generator::getMinP, modificationInfos.getMinActivePower(), "Min active power");
+        if (modificationInfos.getMaxActivePower().getValue() > generator.getMinP()) {
```

Comment:
modificationInfos.getMaxActivePower() can be null ...

Reply:
should we also check if modificationInfos.getMaxActivePower().getValue()>modificationInfos.getMinActivePower().getValue() ?